tutorials/plot_spatial.py

```diff
@@ -11,40 +11,237 @@
+# 5. Compute and plot occupancy
+# 6. Compute 2D and 1D spatial information
 #
 
 ###################################################################################################
 
 # sphinx_gallery_thumbnail_number = 1
 
-import spiketools
+# import auxiliary libraries
+import numpy as np
+import seaborn as sns
+import matplotlib.pyplot as plt
+
+# import all functions from spiketools.spatial
+from spiketools.spatial.position import *
+from spiketools.spatial.occupancy import *
+from spiketools.spatial.utils import *
+from spiketools.spatial.information import *
+
+# import sim_spiketrain_binom to simulate spiketrain
+from spiketools.sim.dist import sim_spiketrain_binom
+
+###################################################################################################
+
+# Set some tracking data
+x_pos = np.linspace(0, 15, 16)
+y_pos = np.array([0, 0, 0.1, 1, 1.5, 1, 1, 2.1, 2, 1, 0, 1, 2, 3, 4, 3.2])
+position = np.array([x_pos, y_pos])
+
+# Set the time width of each position sample
+bin_widths = np.array([1, 1, 1, 2, 1.5, 1, 1, 1, 2, 1, 1, 1, 1, 1, 0.5])
+
+# Set number of spatial bins, 3 x-bins and 5 y-bins
+bins = [3, 5]
+
+# Set timestamp
+timestamps = np.array([0, 1, 2, 3, 5, 6.5, 7.5, 8.5, 9.5, 11.5, 12.5, 13.5, 14.5, 15.5, 16.5, 17])
+
+###################################################################################################
+# 1. Compute and plot distances and speed using position
+# ~~~~~~~~
+#
+# Use x- and y-position to compute distance between two or more points, cumulative distance 
+# traveled and speed
+#
+
+###################################################################################################
+
+# Look at the range of our x and y positions
+ranges = get_pos_ranges(position)
+print(f'The x-position ranges from {ranges[0][0]} to {ranges[0][1]}')
+print(f'The y-position ranges from {ranges[1][0]} to {ranges[1][1]}')
+
+# Plot positions (coordinates marked x are the actual points)
+plt.plot(x_pos, y_pos, '-x', c = 'tab:gray', markersize = 10)
+plt.title('Tracking')
+plt.xlabel('x-position')
+plt.ylabel('y-position')
+plt.legend(['coordinates'])
+
+###################################################################################################
+
+# Compute distance between start and end point
+dist_start_end = compute_distance(x_pos[0], y_pos[0], x_pos[-1], y_pos[-1])
+# Compute distance traveled at each point
+dist_traveled = compute_distances(x_pos, y_pos)
+# Compute total distance traveled
+cumulative_dist_traveled = compute_cumulative_distances(x_pos, y_pos)
+# Compute speed at each point
+bin_widths = np.array([1, 1, 1, 2, 1.5, 1, 1, 1, 2, 1, 1, 1, 1, 1, 0.5])
+speeds = compute_speed(x_pos, y_pos, bin_widths)
+
+###################################################################################################
+
+f, (ax1, ax2, ax3) = plt.subplots(3, 1, sharex=True)
+# Plot distance traveled at each time
+ax1.plot(dist_traveled, '-x', c = 'tab:pink', markersize = 10)
```

Comment:
Some or perhaps all of these plots can probably also use `plot_positions` (?).  
Can you play around with the plots a bit, see what works / looks good, and then we can figure out if we use `plot_positions`everywhere, and/or we might also add a new plot function for other spatial data.